worlds/tww/Macros.py

```diff
@@ -0,0 +1,1203 @@
+            and can_defeat_wizzrobes(state, player)
+            and can_play_winds_requiem(state, player)
+            and can_defeat_darknuts_easily(state, player)
+        )
+        # | (Can Access Item Location "Overlook Island - Cave" & Can Defeat Darknuts Easily)
+        or (
+            state.can_reach_region("Overlook Island Secret Cave", player)
+            and can_defeat_stalfos(state, player)
+            and can_defeat_wizzrobes(state, player)
+            and can_defeat_red_chuchus(state, player)
+            and can_defeat_green_chuchus(state, player)
+            and can_defeat_keese(state, player)
+            and can_defeat_fire_keese(state, player)
+            and can_defeat_morths(state, player)
+            and can_defeat_kargarocs(state, player)
+            and can_play_winds_requiem(state, player)
+            and can_defeat_darknuts_easily(state, player)
+        )
+        # | (Can Access Hyrule)
+        or can_access_hyrule(state, player)
+    )
+
+
+def can_farm_joy_pendants(state: CollectionState, player: int) -> bool:
+    return state.has_all(["Grappling Hook", "Spoils Bag"], player)
+
+
+def can_farm_skull_necklaces(state: CollectionState, player: int) -> bool:
+    return state.has_all(["Grappling Hook", "Spoils Bag"], player)
+
+
+def can_farm_golden_feathers(state: CollectionState, player: int) -> bool:
+    return state.has_all(["Grappling Hook", "Spoils Bag"], player)
+
+
+def can_farm_green_chu_jelly(state: CollectionState, player: int) -> bool:
+    return state.has_all(["Grappling Hook", "Spoils Bag"], player)
+
+
+def can_obtain_15_blue_chu_jelly(state: CollectionState, player: int) -> bool:
+    return (
+        can_get_blue_chu_jelly_from_blue_chuchus(state, player)
+        and (
+            state.has_any(["Hookshot", "Bombs", "Grappling Hook"], player)
+            or can_move_boulders(state, player)
+            or (
+                can_access_secret_cave_entrance_on_shark_island(state, player)
+                and can_fly_with_deku_leaf_outdoors(state, player)
+            )
+            or state.can_reach_region("Cliff Plateau Isles Inner Cave", player)
+            or can_fan_with_deku_leaf(state, player)
+            or can_access_secret_cave_entrance_on_boating_course(state, player)
+        )
+        and state.has("Spoils Bag", player)
+    )
+
+
+def can_defeat_bokoblins(state: CollectionState, player: int) -> bool:
+    return (
+        state.has_any(["Bombs", "Skull Hammer"], player)
+        or has_heros_sword(state, player)
+        or has_heros_bow(state, player)
+    )
+
+
+def can_defeat_green_bokoblins(state: CollectionState, player: int) -> bool:
+    return can_defeat_bokoblins(state, player)
+
+
+def can_defeat_blue_bokoblins(state: CollectionState, player: int) -> bool:
+    return can_defeat_bokoblins(state, player)
+
+
+def can_defeat_pink_bokoblins(state: CollectionState, player: int) -> bool:
+    return can_defeat_bokoblins(state, player)
+
```

Comment:
Since the latter 3 are all the same as the first, I would recommend just deleting the green, blue, and pink entries, especially since they aren't actually used anywhere